include/seqan3/io/alignment_file/header.hpp

```diff
@@ -12,18 +12,54 @@
 
 #pragma once
 
+#include <deque>
 #include <unordered_map>
 #include <vector>
 
-#include <seqan3/core/platform.hpp>
+#include <seqan3/core/metafunction/pre.hpp>
+#include <seqan3/io/alignment_file/detail.hpp>
+#include <seqan3/std/ranges>
 
 namespace seqan3
 {
 
+class alignment_file_format_sam;
+
 //!\brief Stores the header information of alignment files.
 //!\ingroup alignment_file
-struct alignment_file_header
+template <std::ranges::ForwardRange ref_ids_type = std::deque<std::string>>
+class alignment_file_header
 {
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default constructor is defaulted.
+    alignment_file_header() = default;
+    //!\brief Copy construction is defaulted.
+    alignment_file_header(alignment_file_header const &) = default;
+    //!\brief Copy assignment is defaulted.
+    alignment_file_header & operator=(alignment_file_header const &) = default;
+    //!\brief Move construction is defaulted.
+    alignment_file_header(alignment_file_header &&) = default;
+    //!\brief Move assignment is defaulted.
+    alignment_file_header & operator=(alignment_file_header &&) = default;
+    //!\brief Destructor is defaulted.
+    ~alignment_file_header() = default;
+
+    /*!\brief Construct from a range of reference ids which redirects the `ref_ids_ptr` member (non-owning).
+     * \param[in] ref_ids The range over reference ids to redirect the pointer at.
+     */
+    alignment_file_header(ref_ids_type & ref_ids) :
+        ref_ids_ptr{&ref_ids, ref_ids_deleter_noop}
+    {}
+
+    //!\overload
+    alignment_file_header(ref_ids_type && ref_ids) :
+        ref_ids_ptr{new ref_ids_type{std::move(ref_ids)}, ref_ids_deleter_default}
+    {}
+    //!\}
```

Comment:
have you verified that the whole pointer and ownership machinery works if ref_ids_type contains `const`

Reply:
no :sob:

Reply:
it works now?